src/grt/src/fastroute/src/maze.cpp

```diff
@@ -1327,6 +1307,116 @@ float getCost(const int i,
   return cost;
 }
 
+int FastRouteCore::splitEdge(std::vector<TreeEdge>& treeedges,
+                             std::vector<TreeNode>& treenodes,
+                             int n1,
+                             int n2,
+                             int edge_n1n2)
+{
+  const int n2x = treenodes[n2].x;
+  const int n2y = treenodes[n2].y;
+
+  // create new node
+  const int new_node_id = treenodes.size();
+  TreeNode new_node;
+  new_node.x = n2x;
+  new_node.y = n2y;
+  new_node.stackAlias = treenodes[n2].stackAlias;
+
+  // create new edge
+  const int new_edge_id = treeedges.size();
+  TreeEdge new_edge;
+
+  // find one neighbor node different to n1
+  int B1;
+  int edge_n2B1;
+  if (treenodes[n2].nbr[0] == n1) {
+    B1 = treenodes[n2].nbr[1];
+    edge_n2B1 = treenodes[n2].edge[1];
+  } else if (treenodes[n2].nbr[1] == n1) {
+    B1 = treenodes[n2].nbr[0];
+    edge_n2B1 = treenodes[n2].edge[0];
+  } else {
+    B1 = treenodes[n2].nbr[0];
+    edge_n2B1 = treenodes[n2].edge[0];
+  }
```

Comment:
Why the `else if` and the `else` do the same thing? Can they be merged into a single `else`, or should they do different things?